docs: add getting started section (#3818)
aws-powertools/powertools-lambda-typescript: docs/getting-started/navigating-the-toolkit.md

---

> title: Navigating the Toolkit

Reviewer: I like this section to guide people and explain that there are different ways to use. But I am not happy with the title, "Navigating" does not fit here and it's too long. If we move the installation section as the first entry in "Getting Started", this section can be renamed to "Using the Toolkit" or "Usage Patterns", "Implementation Patterns", something in this direction

Author: I agree, I'll name it "Usage Patterns" for now, since this is all there is.

If we add more content in the future we can rename.

description: Getting to know the Powertools for AWS Lambda Toolkit
---

<!-- markdownlint-disable MD043 -->

Powertools for AWS Lambda (TypeScript) is a collection of utilities designed to help you build serverless applications on AWS.

> The toolkit is designed to be modular, so you can pick and choose the utilities you need for your application. Each utility is designed to be used independently, but they can also be used together to provide a complete solution for your serverless applications.

Reviewer: ```suggestion
The toolkit is modular, so you can pick and choose the utilities you need for your application, but also combine them for a complete solution for your serverless applications.
```

Feels a bit redundant, let's simplify it.

## Patterns

Many of the utilities provided can be used with different patterns, depending on your preferences and the structure of your code.

### Class Method Decorator

If you prefer writing your business logic using Object-Oriented Programming (OOP) and TypeScript Classes, the Class Method decorator pattern is a good fit. This approach lets you decorate class methods with Powertools utilities, applying their functionality with minimal code changes.

> This pattern works well when you want to integrate Powertools for AWS Lambda (TypeScript) into an existing codebase without significant refactoring and with no additional runtime dependencies. Note that this approach relies on TypeScript's experimental decorator feature.

Reviewer 2: Should we add the note as an info box? Potentially a reference for how to enable the experimental feature?

Reviewer 2: For example to TypeScript configuration: https://d3tpfv9fcpztxr.cloudfront.net/getting-started/typescript-settings/#typescript-configuration.

> ```ts

Reviewer: Should we extract this example into a snippet as we do with other lambda function examples?

```ts
import { Logger } from '@aws-lambda-powertools/logger';
import { injectLambdaContext } from '@aws-lambda-powertools/logger/middleware';
import { Tracer } from '@aws-lambda-powertools/tracer';
import { captureLambdaHandler } from '@aws-lambda-powertools/tracer/middleware';
import { Metrics } from '@aws-lambda-powertools/metrics';
import { logMetrics } from '@aws-lambda-powertools/metrics/middleware';
import middy from '@middy/core';

const logger = new Logger();
const tracer = new Tracer();
const metrics = new Metrics();

export const handler = middy()
  .use(captureLambdaHandler(tracer))
  .use(injectLambdaContext(logger))
  .use(logMetrics(metrics))
  .handler(async (event: unknown) => {
    // Your business logic here
  });
```

### Functional Approach

If you prefer a more functional programming style, you can use the Powertools for AWS Lambda (TypeScript) utilities directly in your code without decorators or middleware. This approach is more verbose but provides the most control over how the utilities are applied.

> ```ts

Reviewer: Same as above, let's extract it to snippets
